### PR TITLE
Add a note for GroupBy and compute.max_rows

### PR DESCRIPTION
**Pull request by lfdversluis**
This clarifies a question raised in #2072 regarding the impact of max_rows on the size of dataframes/series received by function when using, e.g., `groupby` and `apply`, where the groups can exceed `compute.max_rows` rows.

### CODE REVIEW

**Review by itholic — APPROVED** [review]
Thanks for the improve the documents :)
Seems fine to me.

File: docs/source/user_guide/options.rst [hunk 1]
```diff
@@ -240,7 +240,10 @@ compute.max_rows                1000           'compute.max_rows' sets the limit
                                                When the limit is set, it is executed by the shortcut
```
Comment (HyukjinKwon) [inline docs/source/user_guide/options.rst:240]: I think it's easier to just document that `'compute.max_rows' sets the limit of the current DataFrame.` -> `compute.max_rows' sets the limit of the current Koalas DataFrame.`.

It is documented that `groupby.apply` passes pandas DataFrame so explicitly mentioning it is a Koalas DataFrame will clarify all similar cases.

**Review by HyukjinKwon — CHANGES_REQUESTED** [review]
Can you also update https://github.com/databricks/koalas/blob/e6f90e880ca44d8e73f8480dc4d2e03ae910aeee/databricks/koalas/config.py#L137-L140? This table is supposed to be generated via:

https://github.com/databricks/koalas/blob/e6f90e880ca44d8e73f8480dc4d2e03ae910aeee/databricks/koalas/config.py#L116-L117

File: docs/source/user_guide/options.rst (hunk 1 above)
Comment (lfdversluis) [reply docs/source/user_guide/options.rst:240]: If I understand correctly, even though `compute.max_rows` rows will be loaded into memory, Koalas will still ensure that if a group exceeds `compute.max_rows` rows, it will pass a Pandas DataFrame with all rows the the function(s). I think emphasizing this is an important, too. Any suggestions?

**Comment by lfdversluis** [comment]
I'll fix the lint errors but on Windows I am getting.
```
Traceback (most recent call last):
  File "c:\users\l\appdata\local\programs\python\python37\lib\site-packages\sphinx\config.py", line 327, in eval_config_file
    execfile_(filename, namespace)
  File "c:\users\l\appdata\local\programs\python\python37\lib\site-packages\sphinx\util\pycompat.py", line 88, in execfile_
    exec(code, _globals)
  File "C:\Users\L\Documents\koalas\docs\source\conf.py", line 59, in <module>
    gendoc()
  File "C:\Users\L\Documents\koalas\docs\source\conf.py", line 56, in gendoc
    gendoc.gen_release_notes(source_dir)
  File "C:\Users\L\Documents\koalas\docs\source/../../dev/gendoc.py", line 137, in gen_release_notes
    release_file.write(release_doc)
  File "c:\users\l\appdata\local\programs\python\python37\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u200b' in position 1486: character maps to <undefined>
``` 

I will check this out quickly.

File: docs/source/user_guide/options.rst [hunk 2]
```diff
@@ -270,6 +273,5 @@ plotting.sample_ratio           None           'plotting.sample_ratio' sets the
                                                'plotting.max_rows' option.
 plotting.backend                'plotly'       Backend to use for plotting. Default is plotly.
                                                Supports any package that has a top-level `.plot`
-                                               method. Some options are: [matplotlib, plotly,
```
Comment (lfdversluis) [inline docs/source/user_guide/options.rst:273]: I double checked this because I didn't change a line, but it looks like a previous change in config.py wasn't synced. I figured I could add it in this PR as well.

**Comment by lfdversluis** [comment]
For safety, the hardcoded paths should perhaps be refactored using `os.path.join`, but that is for another PR :)

**Comment by lfdversluis** [comment]
Ah I see Koalas still supports Python 3.5, f-strings were introduced in 3.6. I thought 3.5 was dropped along with [Spark 3.1.1.](https://spark.apache.org/releases/spark-release-3-1-1.html), my bad.

**Comment by lfdversluis** [comment]
@HyukjinKwon please have another look.

File: dev/gendoc.py [hunk 3]
```diff
@@ -19,21 +19,20 @@
...
+import re
 import shutil
 import sys
 from distutils.version import LooseVersion
 from urllib.error import HTTPError
 from urllib.request import urlopen, Request
-import json
-import os
-import re
-import platform
```
Comment (HyukjinKwon) [inline dev/gendoc.py:30]: I would remove unrelated changes here.

File: dev/gendoc.py [hunk 4]
```diff
@@ -19,21 +19,20 @@
...
 
+import databricks.koalas as ks
 import pypandoc
 from pypandoc import pandoc_download
 
-import databricks.koalas as ks
```
Comment (HyukjinKwon) [inline dev/gendoc.py:35]: and PEP8 specifies the import order. I believe the previous import order was correct. Let's revert this change back.

File: docs/source/user_guide/options.rst (hunk 2 above)
Comment (HyukjinKwon) [reply docs/source/user_guide/options.rst:273]: Yeah, sounds good!

File: databricks/koalas/config.py [hunk 5]
```diff
@@ -134,10 +134,12 @@ def validate(self, v: Any) -> None:
...
         doc=(
-            "'compute.max_rows' sets the limit of the current DataFrame. Set `None` to unlimit "
-            "the input length. When the limit is set, it is executed by the shortcut by "
-            "collecting the data into driver side, and then using pandas API. If the limit is "
-            "unset, the operation is executed by PySpark. Default is 1000."
+            "'compute.max_rows' sets the limit of the current Koalas DataFrame. Set `None` to "
+            "unlimit the input length. When the limit is set, it is executed by the shortcut by "
+            "collecting the data into the driver, and then using the pandas API. If the limit is "
+            "unset, the operation is executed by PySpark. Note that this option does not impact "
+            "the sizes of the Pandas DataFrames/Series received by functions of GroupBy. "
```
Comment (HyukjinKwon) [inline databricks/koalas/config.py:141]: I suggested to remove this because other friends like pandas DataFrame/Series via `DataFrame.apply_batch` also don't get affected by this configuration. Elaborating these cases all looks slightly an overkill, and I think "Koalas DataFrame on driver side" clarifies it enough.

**Comment by lfdversluis** [comment]
@HyukjinKwon I understand now what you mean by mentioning explicitly Koalas Dataframe. From this you can derive that other DataFrames are not impacted. I made the changes and rebased.

File: docs/source/user_guide/options.rst [hunk 6]
```diff
@@ -270,6 +270,7 @@ plotting.sample_ratio           None           'plotting.sample_ratio' sets the
...
-                                               pandas_bokeh, pandas_altair].
+                                               method. Known options are: [matplotlib, plotly].
 =============================== ============== =====================================================
+
+
```
Comment (HyukjinKwon) [inline docs/source/user_guide/options.rst:276]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]
Looks good. Thank you @lfdversluis for clarifying this.
